model_stubs/BaseAutoencoder.py

```diff
@@ -0,0 +1,44 @@
+from abc import abstractmethod
+
+from torch import nn
+
+
+'''
+Every implemented Autoencoder should inherit this base class
+'''
+
+class BaseAutoencoder(nn.Module):
+
+    def __init__(self):
+        super(BaseAutoencoder, self).__init__()
+
+    @abstractmethod
+    def encode(self, input):
+        '''
+        This method should call the Encoder part of the model and return the result of the encoder
+        :param input: The input image. Can be single image or batch of images , Tensor of [N x C x H x W]
+        :return: encoder-result
+        '''
+        raise NotImplemented
```

Comment:
Müsste `NotImplementedError` sein oder?

Könntest auch einfach `pass` machen weil die abstractmethod das sowieso nicht zulässt

```suggestion
        raise NotImplementedError
```